src/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -52,6 +54,11 @@ public CommandResult Execute()
                     // Reset the Reporters to the new Console Out and Error.
                     Reporter.Reset();
 
+                    if (!string.IsNullOrEmpty(_workingDirectory))
+                    {
+                        Directory.SetCurrentDirectory(_workingDirectory);
```

Comment:
Up to you. My original design was to allow practically any piece of the command tree to be easily swapped out, so someone could call "dotnet-compile-csc" from bash and they would just have to pass the right arguments to link it up. Basically, there would be no friction in switching languages aside from finding the right arguments to be passed, which could be just copy-pasted from a `dotnet -v` call.

Adding more "types" of options or more mechanisms for delivering those options seems to make things more complicated, IMHO, but we've significantly diverged in overall command structure from where we originally were, so it's possible this approach is more consistent with the overall tool's current design.